tests/test_web_urldispatcher.py

```diff
@@ -407,3 +408,14 @@ async def test_static_absolute_url(aiohttp_client, tmp_path) -> None:
     client = await aiohttp_client(app)
     resp = await client.get('/static/' + str(file_path.resolve()))
     assert resp.status == 403
+
+def test_register_resource(aiohttp_client) -> None:
+    router = UrlDispatcher()
+    path = "/result/"
+    name = "check-for-result"
+    url = URL.build(path=path)
+    resource = PlainResource(url.raw_path, name=name)
+    try:
+        router.register_resource(resource)
+    except ValueError:
+        pytest.fail()
```

Comment:
It's a bad idea to shadow an actual error. the pytest-way is to use
```suggestion
    with pytest.raises(
            ValueError,
            match="^Incorrect route name 'check-for-result', "
            'python keywords cannot be used for route name$',
    ):
        router.register_resource(resource)
```

Reply:
Also, this test doesn't cover negative cases and corner cases. They should be included and the tests should contain both cases with wrong names and correct names. Wrong names should contain cases from both categories — per each if-block.

Reply:
I want to test that the route name "check-for-result" doesn't raise an error, but this way tests that the route name "check-for-result" raises an error.

Reply:
Ah, so in this case just make the call and assert something. Doing mumbo-jumbo with try/except in tests is unidiomatic and should be avoided. The point of the test is to show that something works, not try to handle any errors that may occur — those should be reported by pytest explicitly.

Reply:
Thanks!
I finished fixing.

Reply:
You should've just clicked `Commit suggestion` on the GitHub UI in that comment above. This preserves authorship.